src/Aspire.Hosting/Redis/RedisBuilderExtensions.cs

```diff
@@ -29,18 +29,18 @@ public static IResourceBuilder<RedisContainerResource> AddRedisContainer(this ID
     }
 
     /// <summary>
-    /// Adds a Redis container to the application model. The default image is "redis" and tag is "latest".
+    /// Adds a Redis resource to the application model.
     /// </summary>
     /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
     /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
-    /// <returns>A reference to the <see cref="IResourceBuilder{RedisContainerResource}"/>.</returns>
-    public static IResourceBuilder<RedisResource> AddRedis(this IDistributedApplicationBuilder builder, string name)
+    /// <param name="port">The host port for the redis server.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{RedisResource}"/>.</returns>
+    public static IResourceBuilder<RedisResource> AddRedis(this IDistributedApplicationBuilder builder, string name, int port = 6379)
```

Comment:
Why is the port being added back in for this?

Reply:
I made the other changes by assuming that this extension is to add a resource to an existing redis server that doesn't need to be provisioned in Docker (might be a wrong assumption). So I realized that the `containerPort` should not be hard-coded, but configurable. I assumed that `port` was the one that actually mattered to be configurable. Is that incorrect?

Reply:
I reverted the other changes after reading the PR and issue that introduced this method. I'll talk about it offline.

Reply:
We have two flavors of Redis in the hosting API. `AddRedis(...)` and `AddRedisContainer(...)`. We give folks more control when they use `AddRedisContainer(...)` because when it is deployed it is deployed as a regular container, so controlling ports etc is possible. The intent that the developer is expressing when they call `AddRedis(...)` is that they just want a Redis endpoint and they don't much care how it is configured - we make some decisions for them.

One of the decisions we make is that we randomly assign the proxy port - because when this gets deployed via a deployment tool (e.g. `azd`, it'll end up being a managed service like Azure Cache for Redis). Right now it deploys as a container in ACA, but that is just a point in time thing for preview 1 and 2. We are working to add support for deploying to managed services.

In that context, specifying a port for `AddRedis(...)` doesn't make sense because that configuration value never flows through to the eventual deployed service, so we don't want to set the expectation that developers have the ability to control port assignments when using this particular variant.